### PR TITLE
[FIX] base: fix create option of import for Many2one in One2many

### PR DESCRIPTION
**Pull request by kitan191**
- Install Inventory
- Create an excel file with the following values :
1) location_id | location_dest_id | Operation Type / database ID | Stock moves / product | Stock moves / unit of measure | Stock moves / name
2) Partner Locations/Vendors | WH/Stock | 1 | Non existent product  | Units | Test
- Go to Inventory > Operations > Transfers and import the file
- Check the "Create if doesn't exist" option for "Stock moves / Product" and click on "TEST" button
Import fails because Product does not exist and is not created.

"name_create_enabled_fields" is used in frontend to store which fields have to be created if they don't exist.
The issue comes from the fact that the keys used in "name_create_enabled_fields" contain
the full path to the related field (i.e. move_lines/product_id), but only the final field name is used
when checking "name_create_enabled_fields" in the backend (i.e. product_id).

The prefix from the keys in "name_create_enabled_fields" should be removed when parsing fields from One2many.

opw-2282087

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kitan191** [comment]
ping @nim-odoo

File: odoo/addons/base/models/ir_fields.py [hunk 1]
```diff
@@ -439,6 +439,12 @@ def _str_to_many2many(self, model, field, value):
 
     @api.model
     def _str_to_one2many(self, model, field, records):
+        name_create_enabled_fields = self._context.get('name_create_enabled_fields') or {}
+        relative_name_create_enabled_fields = {}
+        for key in name_create_enabled_fields:
+            prefix = field.name + '/'
+            if key.startswith(prefix):
+                relative_name_create_enabled_fields[key[len(prefix):]] = name_create_enabled_fields[key]
```
Comment (xmo-odoo) [inline odoo/addons/base/models/ir_fields.py:447]: `field` is a constant so `prefix` is a constant, so it should be possible to lift outside the loop, and also might be clearer as a listcomp? e.g.
```suggestion
        prefix = field.name + '/'
        relative_name_create_enabled_fields = {
            k[len(prefix):]: v
            for k, v in name_create_enabled_fields.items()
            if key.startswith(prefix)
        }
```

**Comment by nim-odoo** [comment]
@kitan191 both commits can be merged ;)

**Comment by kitan191** [comment]
robodoo r+

**Pull request closed without merging**